### PR TITLE
Include release notes for version 0.77.0

### PR DESCRIPTION
**Pull request by adamkvitek**
### Security

- [ ] Change has no security implications (otherwise, ping the security team)

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -12,6 +12,53 @@ Welcome to Semgrep release notes. This document provides an overview of the chan
...
+
+#### Highlights
+
+##### Semgrep CLI and Semgrep CI now ignore the same patterns
+
+With this update, Semgrep CLI now ignores the same patterns as the Semgrep CI by default. Find [the default .semgrepignore on GitHub](https://github.com/returntocorp/semgrep/blob/develop/semgrep/semgrep/templates/.semgrepignore). If you want to return to Semgrep’s previous behavior, create an empty `.semgrepignore` file. However, creating a new `.semgrepignore` overrides the default setup.
+
+##### Autofix improvement
+
+An autofix improvement from https://github.com/chair6 from Hashicorp! Big shoutout to them. Fixes several issues (auto fixing multiple things in the same set of lines)
```
Comment (pabloest) [inline docs/release-notes.md:25]:
```suggestion
An autofix improvement from [https://github.com/chair6](https://github.com/chair6) from Hashicorp! Big shoutout to them. Fixes several issues (auto fixing multiple things in the same set of lines)
```

File: docs/release-notes.md [hunk 2]
```diff
@@ -12,6 +12,53 @@ Welcome to Semgrep release notes. This document provides an overview of the chan
...
+
+#### Additions
+
+##### Scala
+
+Semgrep now correctly matches patterns as `List(...)`.
+
+##### `semgrepignore.`
```
Comment (pabloest) [inline docs/release-notes.md:33]:
```suggestion
##### `semgrepignore.`

```

File: docs/release-notes.md [hunk 3]
```diff
@@ -12,6 +12,53 @@ Welcome to Semgrep release notes. This document provides an overview of the chan
...
+
+##### Constant propagation
+
+Constant propagation is now fully a must analysis, if a variable is undefined in some path then it is considered as a non-constant.
+
+##### Dataflow
+
+Dataflow now considers only reachable nodes, which prevents some false-positive or false-negative findings.
+
+##### The `–time` option now includes time spent on processing
```
Comment (pabloest) [inline docs/release-notes.md:54]: Is it `-time` or `--time`?

**Review by pabloest — APPROVED** [review]

File: docs/release-notes.md (hunk 3 above)
Comment (adamkvitek) [reply docs/release-notes.md:55]: `--time`! Will fix it. 

Screenshot 2021-12-21 at 21 12 52

Comment (adamkvitek) [reply docs/release-notes.md]: Fixed.

File: docs/release-notes.md [hunk 4]
```diff
@@ -12,6 +12,54 @@ Welcome to Semgrep release notes. This document provides an overview of the chan
...
+
+With this update, Semgrep CLI now ignores the same patterns as the Semgrep CI by default. Find [the default .semgrepignore on GitHub](https://github.com/returntocorp/semgrep/blob/develop/semgrep/semgrep/templates/.semgrepignore). If you want to return to Semgrep’s previous behavior, create an empty `.semgrepignore` file. However, creating a new `.semgrepignore` overrides the default setup.
+
+##### Autofix improvement
+
+An autofix improvement from [https://github.com/chair6](https://github.com/chair6) from Hashicorp! Big shoutout to them. Fixes several issues (auto fixing multiple things in the same set of lines). This change addresses several issues related to autofix by adding per-file line and column offset tracking, and uses those offsets when making edits to files. The improvement addresses several edge cases in the existing autofix implementation that Semgrep did not handle correctly previously. The addressed issues are the following: [#4428](https://github.com/returntocorp/semgrep/issues/4428), [#3577](https://github.com/returntocorp/semgrep/issues/3577), [#3388](https://github.com/returntocorp/semgrep/issues/3388).
```
Comment (adamkvitek) [inline docs/release-notes.md:25]: @pabloest I tried to blow this up a bit. Please check, not sure if this is so important, but I am glad for any additional information.

Comment (pabloest) [reply docs/release-notes.md:25]: Thanks, @adamkvitek! Looks good to me.

Comment (adamkvitek) [reply docs/release-notes.md:25]: Resolving this comment! :-)

File: docs/release-notes.md (hunk 3 above)
Comment (adamkvitek) [reply docs/release-notes.md]: Resolving this comment! :-)

**Pull request merged**